core/commonKotlin/src/Instant.kt

```diff
@@ -168,6 +169,270 @@ public actual class Instant internal constructor(public actual val epochSeconds:
 
 }
 
+private class UnboundedLocalDateTime(
+    val year: Int,
+    val month: Int,
+    val day: Int,
+    val hour: Int,
+    val minute: Int,
+    val second: Int,
+    val nanosecond: Int,
+) {
+    fun toInstant(offsetSeconds: Int): Instant {
+        val epochSeconds = run {
+            // org.threeten.bp.LocalDate#toEpochDay
+            val epochDays = run {
+                val y = year.toLong()
+                var total = 365 * y
+                if (y >= 0) {
+                    total += (y + 3) / 4 - (y + 99) / 100 + (y + 399) / 400
+                } else {
+                    total -= y / -4 - y / -100 + y / -400
+                }
+                total += ((367 * month - 362) / 12)
+                total += day - 1
+                if (month > 2) {
+                    total--
+                    if (!isLeapYear(year)) {
+                        total--
+                    }
+                }
+                total - DAYS_0000_TO_1970
+            }
+            // org.threeten.bp.LocalTime#toSecondOfDay
+            val daySeconds = hour * SECONDS_PER_HOUR + minute * SECONDS_PER_MINUTE + second
+            // org.threeten.bp.chrono.ChronoLocalDateTime#toEpochSecond
+            epochDays * 86400L + daySeconds - offsetSeconds
+        }
+        if (epochSeconds < Instant.MIN.epochSeconds || epochSeconds > Instant.MAX.epochSeconds)
+            throw DateTimeFormatException(
+                "The parsed date is outside the range representable by Instant (Unix epoch second $epochSeconds)"
+            )
+        return Instant.fromEpochSeconds(epochSeconds, nanosecond)
+    }
+
+    override fun toString(): String = "UnboundedLocalDateTime($year-$month-$day $hour:$minute:$second.$nanosecond)"
+
+    companion object {
+        fun fromInstant(instant: Instant, offsetSeconds: Int): UnboundedLocalDateTime {
```

Comment:
There is only one call to this function, and it passes `0` for `offsetSeconds`.

Reply:
Yes. Do you see any problems with that?